fastapi_mail/config.py

```diff
@@ -16,20 +17,27 @@ class ConnectionConfig(Settings):
     MAIL_SSL: bool = True
     MAIL_DEBUG: conint(gt=-1, lt=2) = 0
     MAIL_FROM: EmailStr
-    MAIL_FROM_NAME: str = None
-    TEMPLATE_FOLDER: Any = None
+    MAIL_FROM_NAME: Optional[str] = None
+    TEMPLATE_FOLDER: Optional[DirectoryPath] = None
     SUPPRESS_SEND: conint(gt=-1, lt=2) = 0
     USE_CREDENTIALS: bool = True
 
-    @validator("TEMPLATE_FOLDER", pre=True)
-    def create_template_engine(cls, v):
-        template_env = None
-        if isinstance(v, str):
-            if not os.path.exists(v):
-                raise TemplateFolderDoesNotExist(
-                    f"{v} is not a valid path to an email template folder")
+    @validator("TEMPLATE_FOLDER")
+    def template_folder_validator(cls, v):
+        """Validate the template folder directory."""
+        if not v:
+            return
+        # need to convert ``PathLike`` object
+        fp = str(v)
+        if not os.path.exists(fp) or not os.path.isdir(fp) or not os.access(fp, os.R_OK):
+            raise TemplateFolderDoesNotExist(
+                f"{v!r} is not a valid path to an email template folder")
+        return v
 
-            if os.path.isdir(v) and os.access(v, os.R_OK) and validate_path(v):
```

Comment:
Why you remove the `validate path` function? This function check for `path traversal` security vulnerability.

Reply:
Hi,

i had issues with it when I instantiated `ConnectionConfig` outside of the parent folder, in the end I had to monkey patch that function to always return `True` otherwise I would not be able proceed.

I'm not very familiar with the issues of `path traversal`, I'm going to look in to handling it with `pathlib` and make a new commit.

Reply:
What are the issues when you instantiate `ConnectionConfig` outside of the parent folder? Why the previous verification wasn't enough?